roles/rhel_subscribe/tasks/main.yml

```diff
@@ -30,7 +30,14 @@
   redhat_subscription:
     username: "{{ rhel_subscription_user }}"
     password: "{{ rhel_subscription_pass }}"
-    autosubscribe: yes
+
+- name: Retrieve the OpenShift Pool ID
+  command: subscription-manager list --available --matches="OpenShift Enterprise, Premium*" --pool-only
```

Comment:
@lhuard1A I like this approach better anyway, there could be some accounts that have multiple entitlements and they may want to specify which one specifically. Also the potential of host consuming multiple entitlements would be bad as well.